packages/react/src/AsyncAutocomplete/AsyncAutocomplete.tsx

```diff
@@ -90,22 +91,23 @@ export function AsyncAutocomplete<T>(props: AsyncAutocompleteProps<T>): JSX.Elem
       const result: T[] = [];
       for (const value of values) {
         let item = optionsRef.current?.find((option) => option.value === value)?.resource;
-        if (!item) {
+        if (!item && creatable !== false) {
           item = (onCreate as (input: string) => T)(value);
         }
-        result.push(item);
+
+        if (item) result.push(item);
       }
       onChange(result);
     },
-    [onChange, onCreate]
+    [creatable, onChange, onCreate]
   );
 
   const handleKeyDown = useCallback(
     (e: React.KeyboardEvent): void => {
       if (e.key === 'Enter') {
         if (!timerRef.current && !abortControllerRef.current) {
           killEvent(e);
-          if (optionsRef.current && optionsRef.current.length > 0) {
+          if (optionsRef.current && optionsRef.current.length > 0 && creatable !== false) {
```

Comment:
this prevents `Enter` key from overriding Mantine Multiselect behavior when selecting from dropdown. This was causing unnecessary side effect where it navigates to first item of multiselect drop down.